### PR TITLE
Frankreed/remove user assertion failed fix

### LINKED ISSUE #294: Removing a user doesn't work
Getting this error:  ERROR  [2024-07-21T10:20:04.625Z]  @firebase/firestore: Firestore (10.12.3): FIRESTORE (10.12.3) INTERNAL ASSERTION FAILED: Unexpected state
Screenshot 2024-07-21 at 5 22 52 PM

### PR DESCRIPTION
**Pull request by FrankreedX**
closes #294

### CODE REVIEW

File: dbOperations/removeUser.js [hunk 1]
```diff
@@ -13,48 +14,46 @@ import { getPfpName } from "~/Utility";
...
-      //Remove user from user table where UID == userID
-      const userRef = doc(db, "teams", teamId, "users", userId);
+    //Remove user from user table where UID == userID
+    const userRef = doc(db, "teams", teamId, "users", userId);
 
-      await transaction.delete(userRef);
+    await deleteDoc(userRef);
 
-      await removePfp(getPfpName(teamId, userId));
+    await removePfp(getPfpName(teamId, userId));
```
Comment (Gehrkej) [inline dbOperations/removeUser.js:52]: I think this is throwing an error, the user gets deleted but an error message keeps popping up:
image

File: app/content/team/users/[user]/index.js [hunk 2]
```diff
@@ -319,8 +319,17 @@ function Index() {
               ]);
               navigation.goBack();
             } catch (e) {
-              console.log("Error removing user:", e);
-              showDialog("Error", getErrorString(e));
+              if (e.code === "storage/object-not-found") {
```
Comment (Gehrkej) [inline app/content/team/users/[user]/index.js:322]: looks like you were trying to have a temp fix here, might be better to get to the root cause so we can validate it is working for the long run

File: dbOperations/removeUser.js
Comment (Gehrkej) [inline dbOperations/removeUser.js:1]: This looks like it is relatively the same except it removes the transaction no? Just for clarification.

**Review by Gehrkej — CHANGES_REQUESTED** [review]
Seems like this bug still needs some work, I might try to take a look at it after completing some other PR reviews although I think I need this address so I can properly test PR #289

File: dbOperations/removeUser.js
Comment (FrankreedX) [reply dbOperations/removeUser.js:1]: yes. the error you received is meant to be handled in app/content/team/users/[user]/index.js with the if statement you highlighted.

**Pull request closed without merging**